cartographer_ros/cartographer_ros/assets_writer.h

```diff
@@ -17,20 +17,48 @@
 #include <string>
 #include <vector>
 
+#include "cartographer/common/configuration_file_resolver.h"
+#include "cartographer/io/points_processor_pipeline_builder.h"
+#include "cartographer/mapping/proto/pose_graph.pb.h"
+#include "cartographer/mapping/proto/trajectory_builder_options.pb.h"
+
 #ifndef CARTOGRAPHER_ROS_ASSETS_WRITER_H_
 #define CARTOGRAPHER_ROS_ASSETS_WRITER_H_
 
 namespace cartographer_ros {
 
-// Configures a point processing pipeline and pushes the points from the bag
-// through it.
-void RunAssetsWriterPipeline(const std::string& pose_graph_filename,
-                             const std::vector<std::string>& bag_filenames,
-                             const std::string& configuration_directory,
-                             const std::string& configuration_basename,
-                             const std::string& urdf_filename,
-                             const std::string& output_file_prefix,
-                             bool use_bag_transforms);
+class AssetsWriter {
+ public:
+  // Configures a point processing pipeline.
+  AssetsWriter(const std::string& pose_graph_filename,
+               const std::vector<std::string>& bag_filenames,
+               const std::string& configuration_directory,
+               const std::string& configuration_basename,
+               const std::string& urdf_filename,
+               const std::string& output_file_prefix,
+               const bool use_bag_transforms);
+
+  // Pushes the points from the bag through the point processing pipeline.
+  void Run();
+
+ private:
+  void Init(const std::string& pose_graph_filename,
+            const std::string& configuration_directory,
+            const std::string& configuration_basename,
+            const std::string& output_file_prefix);
+
+  std::vector<std::string> bag_filenames_;
+  std::string urdf_filename_;
+  bool use_bag_transforms_;
+  std::vector<::cartographer::mapping::proto::Trajectory> all_trajectories_;
+  std::unique_ptr<::cartographer::mapping::proto::PoseGraph> pose_graph_;
```

Comment:
Why `unique_ptr`? Can't you just move into the protos like before?

Reply:
You're right, it's not needed. I will change it.